assets/src/edit-story/components/panels/design/preset/colorPreset/colorPresetActions.js

```diff
@@ -17,241 +17,64 @@
-import { Icons } from '@web-stories-wp/design-system';
 
 /**
  * Internal dependencies
  */
 import { useStory } from '../../../../../app/story';
-import { Select } from '../../../../form';
-import { findMatchingColor } from '../utils';
 import { SAVED_COLOR_SIZE } from '../../../../../constants';
-import { focusStyle } from '../../../shared';
 import ColorGroup from './colorGroup';
 import useApplyColor from './useApplyColor';
 
-const LOCAL = 'local';
-const GLOBAL = 'global';
-const COLOR_GAP = 6;
-
-const ActionsWrapper = styled.div`
-  text-align: center;
-  border-top: 1px solid ${({ theme }) => theme.colors.divider.primary};
-  padding: 8px 16px 20px 16px;
-`;
-
-const AddColorPreset = styled.button`
-  background: transparent;
-  border: none;
-  color: ${({ theme }) => theme.colors.fg.secondary};
-  cursor: pointer;
-  padding: 0;
-  line-height: 20px;
-  svg {
-    color: ${({ theme }) => theme.colors.fg.secondary};
-    width: 32px;
-    height: 32px;
-  }
-  ${focusStyle};
-`;
-
-const CtaWrapper = styled.div`
-  font-size: 14px;
-  line-height: 32px;
-  color: ${({ theme }) => theme.colors.fg.tertiary};
-
-  svg {
-    color: ${({ theme }) => theme.colors.standard.white};
-    width: 32px;
-    height: 32px;
-    vertical-align: bottom;
-  }
-`;
+const COLOR_GAP = 4;
+const MAX_ROWS = 2;
 
 const ColorsWrapper = styled.div`
-  max-height: ${SAVED_COLOR_SIZE * 3 + 2 * COLOR_GAP}px;
-  padding: 4px;
-  margin: 6px 0 0 -4px;
+  max-height: ${SAVED_COLOR_SIZE * MAX_ROWS + COLOR_GAP * (MAX_ROWS - 1)}px;
   width: calc(100% + 8px);
   overflow-x: hidden;
   overflow-y: auto;
 `;
 
-const DropDownWrapper = styled.div`
-  text-align: left;
-  position: relative;
-  flex: 1 1 100%;
-`;
-
-const HeaderRow = styled.div`
-  display: flex;
-`;
-
-const ButtonWrapper = styled.div`
-  text-align: end;
-  flex-grow: 1;
-`;
-
-const Strong = styled.span`
-  font-size: 24px;
-  color: ${({ theme }) => theme.colors.standard.white};
-`;
-
 /**
  * @param {Object} properties Properties.
- * @param {Object} properties.color Current color.
  * @param {Function} properties.pushUpdate Update function.
- * @param {Function} properties.onAction Function called when user initiates an action.
  * @return {*} Element.
  */
-function ColorPresetActions({ color, pushUpdate, onAction }) {
-  const [showLocalColors, setShowLocalColors] = useState(true);
-  const {
-    selectedElements,
-    currentStoryStyles,
-    globalStoryStyles,
-    updateStory,
-  } = useStory(
+function ColorPresetActions({ pushUpdate }) {
+  const { globalStoryStyles } = useStory(
     ({
       state: {
-        selectedElements,
-        story: { globalStoryStyles, currentStoryStyles },
+        story: { globalStoryStyles },
       },
-      actions: { updateStory },
-    }) => ({
-      selectedElements,
-      globalStoryStyles,
-      currentStoryStyles,
-      updateStory,
-    })
+    }) => ({ globalStoryStyles })
   );
 
   const { colors: globalColors } = globalStoryStyles;
```

Comment:
Use `colors: savedColors` directly?

Reply:
Yeah sure, I was just cleaning this up roughly removing all the unnecessary, not actually reviewing the code to see if it could be optimized. I'll give this an extra overhaul now that it's been greatly simplified.